budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -168,3 +174,141 @@ def test_other_parties(self):
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        response_data = response.data["results"][0]
+        self.assertEqual(number_of_projects, 1)
+        self.assertEqual(response_data["name"], project.name)
+        self.assertEqual(response_data["province"], project.province)
+        self.assertEqual(
+            response_data["local_municipality"], project.local_municipality
+        )
+        self.assertEqual(
+            response_data["district_municipality"], project.district_municipality
+        )
+
+    def test_filter_by_province(self):
+        projects = ProvInfraProject.objects.filter(province=self.province)
+
+        data = {"province": self.province}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, projects.count())
+
+    def test_filter_by_status(self):
+        projects = ProvInfraProject.objects.filter(status=self.status)
+
+        data = {"status": self.status}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, projects.count())
+
+    def test_filter_by_funding_source(self):
+        projects = ProvInfraProject.objects.filter(primary_funding_source=self.source)
+
+        data = {"primary_funding_source": self.source}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, projects.count())
+
+    def test_search_by_project_name(self):
+        data = {"search": self.name}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.name)
+
+    def test_search_by_municipality(self):
+        data = {"search": self.municipality}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.municipality)
+
+    def test_search_by_province(self):
+        data = {"search": self.province}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.province)
+
+    def test_search_by_contractor(self):
+        data = {"search": self.contractor}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.contractor)
+
+    def test_project_detail_content(self):
```

Comment:
please split this out to another test class - you can use the existing infrastructure projects fixture.

This is a test of the infra project detail HTML view so it doesn't belong in the API test class.

Reply:
test moved into a new test class.

https://github.com/vulekamali/datamanager/pull/190/commits/88f7671b86e5c76bfde46e5847018f88492a6ac7